### PR TITLE
Pass ddof=0 to np.cov in KernelDensity.fit

### PR DESCRIPTION
**Pull request by ogrisel**
Attempt to improve `sample_weight` equivalence properties for scikit-learn/scikit-learn#27971.

### CODE REVIEW

**Comment by ogrisel** [comment]
Actually let me move this to draft mode because it does not yet pass for all admissible values of `global_random_seed`. I am working on a fix.

**Comment by Charlie-XIAO** [comment]
@ogrisel I think this should target the main repo instead of my fork :)

**Comment by ogrisel** [comment]
> @ogrisel I think this should target the main repo instead of my fork :)

This was intentional to only see a small diff and not to have two competing open PRs with 99% of shared commits.

**Comment by Charlie-XIAO** [comment]
Ah I see now I didn't notice your comment in my other PR :)

File: sklearn/neighbors/tests/test_kde.py [hunk 1]
```diff
@@ -204,9 +204,9 @@ def test_kde_sample_weights(algorithm, metric, d):
     scores_ref_sampling = kde.score_samples(test_points)
     sample_ref_sampling = kde.sample(random_state=1234)
     assert_allclose(
-        np.exp(scores_weight), np.exp(scores_ref_sampling), atol=1e-8, rtol=1e-2
+        np.exp(scores_weight), np.exp(scores_ref_sampling), atol=1e-8, rtol=1e-6
```
Comment (Charlie-XIAO) [inline sklearn/neighbors/tests/test_kde.py:207]: It looks like `atol` can be removed with the fix.

File: sklearn/neighbors/_kde.py [hunk 2]
```diff
@@ -220,8 +220,7 @@ def fit(self, X, y=None, sample_weight=None):
             sample_weight = _check_sample_weight(
                 sample_weight, X, dtype=np.float64, ensure_non_negative=True
             )
-            normalized_sample_weight = sample_weight / sample_weight.sum()
-            n_effective_samples = 1 / np.sum(normalized_sample_weight**2)
```
Comment (ogrisel) [inline sklearn/neighbors/_kde.py:224]: This is necessary to make the `assert_allclose(scores_weight, scores_repetition, **tols)` test pass when using "scott" or "silverman" bandwith. However, this alone is not enough to make the other assertions in the same test pass.

**Comment by ogrisel** [comment]
This is still not yet ready to merge. I broke other tests.

**Comment by ogrisel** [comment]
The main remaining problem is that I broke the equivalence with `gaussian_kde` (in the unweighted case) by passing `ddof=0`...

**Comment by Charlie-XIAO** [comment]
Besides the `neff` problem, I'm also concerned whether we should really assume sample results are close? That's just a random sample - I don't think that means equivalent sampling even with the same seed if I'm not mistaken.
